test/curve_analysis/test_baseclass.py

```diff
@@ -629,12 +735,52 @@ def test_multi_composite_curve_analysis(self, plot, gen_figures, n_figures):
         self.assertExperimentDone(result)
         amps = result.analysis_results("amp")
 
+        fit_A = expdata.artifacts("fit_summary").data["group_A"]
+        self.assertEqual(
+            fit_A.model_repr,
+            {"m1": "amp * cos(2 * pi * freq * x) + b", "m2": "amp * sin(2 * pi * freq * x) + b"},
+        )
+        self.assertEqual(fit_A.init_params, group_A_p0)
+        self.assertAlmostEqual(fit_A.params["amp"], amp1, delta=0.1)
+        self.assertAlmostEqual(fit_A.params["freq"], freq1, delta=0.1)
+        self.assertAlmostEqual(fit_A.params["b"], b1, delta=0.1)
+
+        fit_B = expdata.artifacts("fit_summary").data["group_B"]
+        self.assertEqual(
+            fit_B.model_repr,
+            {"m1": "amp * cos(2 * pi * freq * x) + b", "m2": "amp * sin(2 * pi * freq * x) + b"},
+        )
+        self.assertEqual(fit_B.init_params, group_B_p0)
+        self.assertAlmostEqual(fit_B.params["amp"], amp2, delta=0.1)
+        self.assertAlmostEqual(fit_B.params["freq"], freq2, delta=0.1)
+        self.assertAlmostEqual(fit_B.params["b"], b2, delta=0.1)
+
+        table_subset = expdata.artifacts("curve_data").data.filter(series=series, category=category)
```

Comment:
How the setting for the figure generation impacts the ScatterTable?

Reply:
It doesn't, I just tested both here since it's the integration test for composite curve analysis.